src/main/java/dk/bec/polonez/reservationsystem/controller/UserController.java

```diff
@@ -1,60 +1,53 @@
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-import org.springframework.web.server.ResponseStatusException;
+import dk.bec.polonez.reservationsystem.dto.userDto.*;
+import dk.bec.polonez.reservationsystem.service.UserService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
 
 @RestController
-@RequestMapping("/api/user/")
+@RequestMapping("/api/users/")
 public class UserController {
 
-    private final UserRepository userRepository;
+    private final UserService userService;
 
-    private final RoleRepository roleRepository;
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
 
-    private final PasswordEncoder encoder;
+    @PostMapping()
+    public SignupResponse registerUser(@RequestBody SignupRequest signUpRequest) {
+        return userService.createUser(signUpRequest);
+    }
 
-    public UserController(UserRepository userRepository, RoleRepository roleRepository, PasswordEncoder encoder) {
-        this.userRepository = userRepository;
-        this.roleRepository = roleRepository;
-        this.encoder = encoder;
+    @PutMapping()
+    public UpdateResponse updateUser(@RequestBody UpdateRequest updateRequest) {
+        return userService.updateUser(updateRequest);
     }
 
-    @PostMapping("signup")
-    public SignupResponse registerUser(@RequestBody SignupRequest signUpRequest) {
-        if (userRepository.existsByUsername(signUpRequest.getUsername())) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Error: Username is already taken!");
-        }
-
-        if (userRepository.existsByEmail(signUpRequest.getEmail())) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Error: Email is already in use!");
-        }
-
-        String strRole = signUpRequest.getRole().toUpperCase();
-        Role role = roleRepository.findByName(strRole)
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Role not found"));
-
-        User user = User.builder()
-                .username(signUpRequest.getUsername())
-                .password(encoder.encode(signUpRequest.getPassword()))
-                .email(signUpRequest.getEmail())
-                .role(role)
-                .build();
-
-        User savedUser = userRepository.save(user);
-        return SignupResponse.builder()
-                .username(savedUser.getUsername())
-                .email(savedUser.getEmail())
-                .build();
+    @GetMapping()
+    public List<ProfileResponse> getAll() {
+        return userService.getAllUsers();
+    }
+
+    @GetMapping("{id}")
+    public ProfileResponse getById(@PathVariable Long id) {
+        return userService.getById(id);
+    }
+
+    @PutMapping("block/{id}")
```

Comment:
co wiecej wg REST-a w ogole block nie powinno sie znalezc w path-ie -> w request body powinienes przekazac nazwe pola, ktora ma sie zmienic i na jaka wartosc

powinien to byc Patch, nie put -> jesli korzystasz z put-a to powinienes przekazac cala reprezentacje zasobu, w przypadku patch, mozesz przekazac konkretne pole, ktore ma zostac zupdetowane